dns_sd_windows.c

```diff
@@ -202,14 +204,17 @@ int dnssd_find_hosts(struct dns_sd_discovery_data **ddata)
 
 	IIO_DEBUG("DNS SD: Start service discovery.\n");
 
-	*ddata = zalloc(sizeof(**ddata));
-	if (!*ddata)
+	d = zalloc(sizeof(*d));
+	if (!d)
 		goto out_wsa_cleanup;
 
+	d->lock = iio_mutex_create();
+	if (!d->lock)
+		goto out_destroy_lock;
```

Comment:
This goto looks fishy.

Reply:
yes - that was wrong - should be fixed now. Thanks